packages/amplify_datastore/android/src/main/kotlin/com/amazonaws/amplify/amplify_datastore/DataStoreHubEventStreamHandler.kt

```diff
@@ -0,0 +1,225 @@
+                                var modelName = (outboxMutationEnqueued.element.model as SerializedModel).modelName as String
+                                var res = FlutterOutboxMutationEnqueuedEvent(
+                                        hubEvent.name,
+                                        modelName,
+                                        outboxMutationEnqueued.element
+                                )
+                                sendEvent(res.toValueMap())
+                            } else {
+                                LOG.error("Element is not an instance of SerializedModel.")
+                                sendError(hubEvent.name)
+                            }
+                        } catch (e: Exception) {
+                            LOG.error("Failed to parse and send outboxMutationEnqueued event: ", e)
+                            sendError(hubEvent.name)
+                        }
+
+                    }
+                    DataStoreChannelEventName.OUTBOX_MUTATION_PROCESSED.toString() -> {
+                        try {
+                            var outboxMutationProcessed = hubEvent.data as OutboxMutationEvent<*>
+                            if (outboxMutationProcessed.element.model is SerializedModel) {
+                                var modelName = (outboxMutationProcessed.element.model as SerializedModel).modelName as String
+                                var res = FlutterOutboxMutationProcessedEvent(
+                                        hubEvent.name,
+                                        modelName,
+                                        outboxMutationProcessed.element
+                                )
+                                sendEvent(res.toValueMap())
+                            } else {
+                                LOG.error("Element is not an instance of SerializedModel.")
+                                sendError(hubEvent.name)
+                            }
+                        } catch (e: Exception) {
+                            LOG.error("Failed to parse and send outboxMutationProcessed event: ", e)
+                            sendError(hubEvent.name)
+                        }
+                    }
+                    DataStoreChannelEventName.OUTBOX_STATUS.toString() -> {
+                        try {
+                            var outboxEvent = hubEvent.data as OutboxStatusEvent
+                            var res = FlutterOutboxStatusEvent(hubEvent.name, outboxEvent.isEmpty)
+                            sendEvent(res.toValueMap())
+                        } catch (e: Exception) {
+                            LOG.error("Failed to parse and send outboxStatus event: ", e)
+                            sendError(hubEvent.name)
+                        }
+                    }
+                    else -> {
+                        LOG.error("Unrecognized DataStoreHubEvent")
+                        sendError(hubEvent.name)
+                    }
```

Comment:
lets discuss if this should be an error event or a warning logged with an event just with the name?

Reply:
I would think it's probably more of a two-way door to NOT send an error right now?